Sources/TuistDependencies/DependenciesController.swift

```diff
@@ -191,20 +191,29 @@ public final class DependenciesController: DependenciesControlling {
 
 extension TuistCore.DependenciesGraph {
     public func merging(with other: Self) throws -> Self {
-        let mergedExternalDependencies = try other.externalDependencies.reduce(into: externalDependencies) { result, entry in
-            if let alreadyPresent = result[entry.key] {
-                throw DependenciesControllerError.duplicatedDependency(entry.key, alreadyPresent, entry.value)
-            }
+        let mergedExternalDependencies = try other.externalDependencies.reduce(into: externalDependencies) { result1, entry1 in
```

Comment:
I think that doing 
```suggestion
        let mergedExternalDependencies = try other.externalDependencies.forEach { platrform, otherExternalDependencies in
```
instead of using the reducer initializing into with self would greatly simplify this code

Reply:
Simplified the merging approach